dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,99 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::IndexSelect(
+    int64_t dim, torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  if (dim == 0) {  // row
```

Comment:
The logic of dim = 0 and 1 are very similar. You can merge them for simplification. Basically, they all get the CSR/CSC/COO matrices, call DGL APIs and convert the result to sparse matrix. You only need to add some transpose operators according to the dim.